### PR TITLE
Allow value-less keys, fixes #18

### LINKED ISSUE #18: Support `begin_src` keys without values
I recently started to use `sagemath` via [ob-sagemath](https://github.com/sagemath/ob-sagemath) in some of my Org files. It needs to open a sage session in the background to run the code and requires the `:session` annotation for each block, for example:
```
Some text.
#+begin_src sage :session
φ = var('φ')
#+end_src
Some more text.
#+begin_src sage :session
print(φ)
#+end_src
where both of course run in the same session.
```
For `ntangle` the `:session` doesn't really matter of course. In the context I don't even care about actually tangling these into some file (be it a `.py` file or `.sage`).

The issue however is that this isn't allowed in `ntangle` at the moment. We raise an exception here:

https://github.com/OrgTangle/ntangle/blob/master/src/ntangle.nim#L407

I'm not sure whether there's a deeper reason that this isn't supported currently. I've written the following patch for myself locally:

```
diff --git a/src/ntangle.nim b/src/ntangle.nim
index 36a151b..158f83a 100644
--- a/src/ntangle.nim
+++ b/src/ntangle.nim
@@ -400,16 +400,20 @@ proc getHeaderArgs(file: string, line: string, lnum: int): LangAndArgs =
     if startHeaderArgs >= 2:
       headerArgsRaw = spaceSepParts[startHeaderArgs .. spaceSepParts.high]
   if haType != haNone:
-    #echo headerArgsRaw
+    var lastKey = false
     for i, h in headerArgsRaw:
       if h.len >= 2 and h[0] == ':':
-        if i == headerArgsRaw.high:
-          raise newException(OrgError, fmt("The header args are ending with a ':key' which is not valid. Found {headerArgsRaw}"))
+        if i == headerArgsRaw.high or lastKey:
+          headerArgPair = h[1 .. h.high] & " " & "<none>"
+          headerArgs.add headerArgPair
+          continue
         if headerArgPair != "":
           headerArgs.add(headerArgPair)
         headerArgPair = h[1 .. h.high]
         #echo fmt"{i} - {headerArgPair}"
+        lastKey = true
       else:
+        lastKey = false
         headerArgPair = headerArgPair & " " & h.strip(chars = {'"'})
         #echo fmt"{i} - {headerArgPair}"
         if i == headerArgsRaw.high:
```

As it's a bit hacky and I don't understand the ramifications in the broader context, I thought it's better to open an issue about it first.

### PR DESCRIPTION
**Pull request by Vindaar**
This addresses issue #18 by allowing keys without a value. The exception remains `:tangle` (as that is the main use case for the parsing is for `:tangle` related options), where we continue to raise an exception if no value is provided.

I changed the parsing logic to properly separate keys and values.

Are the files generated by the tests all committed?

### CODE REVIEW

**Comment by kaushalmodi** [comment]
Thanks for quickly coming up with a PR!

> Are the files generated by the tests all committed?

Yes.

I don't have a good way of testing except that the git diff should return nothing if a test passes. As this PR has a mkdirp, you will need to add an `rm -rf` to delete the directory that that test is going to create.

I am not very proud of the testing style :| .. but it kind of works ..

https://github.com/OrgTangle/ntangle/blob/142daae36d39cffdaff8db9eb509bdc99d8e3b71/tests/test.sh#L29-L30

**Comment by Vindaar** [comment]
> I am not very proud of the testing style :neutral_face: .. but it kind of works ..

Well, as long as it works, heh.

**Comment by kaushalmodi** [comment]
Thanks for this PR!

**Comment by kaushalmodi** [comment]
I think I merged this PR a bit too soon.. if a key's value has double-quotes, those are treated literally.

So this will generate a file named `"file_permissions_600_override_shebang"`, literally with the double quotes:

```
#+begin_src shell :tangle "file_permissions_600_override_shebang" :tangle-mode (identity #o600) :shebang "#!/usr/bin/env bash"
echo "permissions of this file will be 600 even though :shebang is used"
#+end_src
```

Investigating ..

---

**Update**: Fixed

File: src/ntangle.nim [hunk 1]
```diff
@@ -400,20 +400,24 @@ proc getHeaderArgs(file: string, line: string, lnum: int): LangAndArgs =
...
-        headerArgPair = h[1 .. h.high]
-        #echo fmt"{i} - {headerArgPair}"
+      if h.len >= 2 and h[0] == ':': # this is a key
+        if notEmpty(keyVal): # add current `keyVal` if anything
+          headerArgs.add keyVal
+          keyVal = (key: "", val: "")
+        keyVal.key = h[1 .. h.high]
+        lastKey = true
       else:
-        headerArgPair = headerArgPair & " " & h.strip(chars = {'"'})
```
Comment (kaushalmodi) [inline src/ntangle.nim:413]: @Vindaar ok.. this part got left out: 

```
h.strip(chars = {'"'})
```

Comment (Vindaar) [reply src/ntangle.nim:413]: Ohh, thanks. Guess I wasn't paying attention, oops!

**Pull request merged**